modAion/src/org/aion/zero/db/AionContractDetailsImpl.java

```diff
@@ -174,17 +200,17 @@ public void decode(byte[] rlpCode) {
     }
 
     @Override
-    public Map<DataWord, DataWord> getStorage(Collection<DataWord> keys) {
-        Map<DataWord, DataWord> storage = new HashMap<>();
+    public Map<IDataWord, IDataWord> getStorage(Collection<IDataWord> keys) {
+        Map<IDataWord, IDataWord> storage = new HashMap<>();
         if (keys == null) {
             throw new IllegalArgumentException("Input keys can't be null");
         } else {
-            for (DataWord key : keys) {
-                DataWord value = get(key);
+            for (IDataWord key : keys) {
+                IDataWord value = get(key);
 
                 // we check if the value is not null,
                 // cause we keep all historical keys
-                if (value != null) {
+                if ((value != null) && (!value.isZero())) {
```

Comment:
it seemed to be that a value of all zeros is equivalent to null in the db and I wanted to make this consistent. I could be wrong here, but in TrieImpl, which receives the new data word, if value is all zeros it interprets this as a request to remove the corresponding key, rather than allowing a zero array to be a valid entry

Reply:
Ok. Please add an ` @apiNote` comment to the Java doc for `IDataWord#isZero()` that the zero value is interpreted as `null`.

Reply:
done